packages/ruby/lib/readme/metrics.rb

```diff
@@ -41,5 +41,41 @@ def call(env)
 
       [status, headers, body]
     end
+
+    private
+
+    def validate_options(options)
+      raise ConfigurationError, "Missing API Key" if options[:api_key].nil?
+
+      options[:reject_params]&.each do |param|
+        unless param.is_a?(String)
+          raise ConfigurationError, "reject_params option must be an array of strings"
+        end
+      end
+
+      options[:allow_only]&.each do |param|
+        unless param.is_a?(String)
+          raise ConfigurationError, "allow_only option must be an array of strings"
+        end
+      end
+
+      if options[:buffer_length] && !options[:buffer_length].is_a?(Integer)
+        raise ConfigurationError, "buffer_length must be an Integer"
+      end
+
+      if options[:development] && !is_a_boolean?(options[:development])
+        raise ConfigurationError, "development option must be a boolean"
+      end
+    end
+
+    def is_a_boolean?(arg)
+      arg.is_a?(TrueClass) || arg.is_a?(FalseClass)
+    end
+  end
+
+  class ConfigurationError < StandardError
+    def initialize(msg)
+      super(msg)
+    end
```

Comment:
This is the same as not defining a constructor right? In both cases it just calls the superclass's implementation.

Common ways of defining errors are:

```ruby
class ConfigurationError < StandardError; end

# or

ConfigurationError = Class.new(StandardError)
```